### DISCUSSION TITLE
Support for non-platform JpegXL decode (and maybe AVIF)

### DISCUSSION THREAD

**Discussion by Quackdoc** [kodjodevf/mangayomi#90] [Ideas]
Im not sure when it is appropriate to open a discussion vs or an issue ticket so I opted for discussion

JpegXL is next generation image format that is considerably smaller then it's competition, For lossless it manages to typically be smaller then PNG for comics and webcomics. And for lossy, You can directly transcode jpg images and the result will 90% of the time be smaller, 9% of the time, and 1% of the time it may loose on some edge cases.

I myself transcode my manga to JXL since it can save a lot of space, However it is possible that since Apple devices now natively support JXL, services may decide to start serving JXL images when possible.

I noticed that this project uses rinf for various things, As a potential way to add support for this (and more image formats in the future like AVIF) Im wondering if it would be viable to have a image decoder plugin in mangayomi using rinf. This should fairly easily allow building against things like `jxl-oxide`, `image-rs`, and other potential image formats down the line without relying on the `image` package or relying on native platform support.

As an aside, I think it would also be neat to have a similar setup for compressing images (as an aside, or in addition to cbz compression)

**Comment by kodjodevf** [comment]
Good idea, by the way I am a beginner in rust I am trying to use the crates that I need, you already have an idea of what needs to be implemented if you want you can make a pull request concerning these subjects that will be good for me of great help.

**Reply by Quackdoc** [reply]
Im not sure ill have enough time to look at this any time soon, but ill put it on a list of things to take a gander at
